### PR TITLE
Prefer per-profile API keys for default LLM profile

### PR DESCRIPTION
**Pull request by enyst**
Fixes a startup UX footgun where a user could set a per-profile override API key (openhands.llmProfileApiKey.<profileId>) but still get defaulted to an unrelated profile when openhands.llm.profileId was unset.

- Default profile selection now considers per-profile override keys first, then provider-global keys, then falls back to sonnet-45.
- Adds unit coverage and updates docs (llm_profiles.md + settings_prd.md).

## Summary by CodeRabbit

* **New Features**
  * Deterministic startup selection and global persistence of a default LLM profile when none is set; best-effort selection favors per-profile keys, then provider keys, then a fallback.
  * Webview now receives updated LLM profile and status info more reliably at startup.

* **Behavior Changes**
  * Explicit profileId updates now clear workspace overrides and persist at the global level.

* **Documentation**
  * Docs updated to describe startup selection, persistence, and revised secrets model.

* **Tests**
  * Added tests ensuring per-profile keys take precedence over provider keys.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by enyst** [comment]
Addressed Gemini inline review:
- Default profile selection no longer uses a hardcoded per-profile key list; it now discovers profiles via @openhands/agent-sdk-ts listProfiles() and checks for matching openhands.llmProfileApiKey.<profileId>, so custom profiles work too.
- Added a unit test covering a custom profile per-profile key.

**Comment by enyst** [comment]
Follow-up on your request (ensure explicit user selection always wins):
- Fixed a real edge case where setting openhands.llm.profileId in VS Code Settings at workspace/workspace-folder scope could be ignored (we previously only read the global value).
- VscodeSettingsAdapter now reads profileId from workspace overrides, but when the webview selects a profile it clears those overrides and persists globally to prevent snap-back.
- ensureConversationAndConnection now also posts llmProfilesUpdated so the dropdown reflects Settings UI changes even without a webview reload.

File: src/settings/SettingsManager.ts [hunk 1]
```diff
@@ -173,6 +182,10 @@ export class SettingsManager {
       return typeof envValue === 'string' && envValue.trim().length > 0;
     };
 
+    for (const entry of DEFAULT_LLM_PROFILE_ID_BY_PROFILE_API_KEY) {
+      if (await hasSecret(entry.secretKey)) return entry.profileId;
+    }
```
[the remark this answers is not in the document]
Comment (enyst) [reply src/settings/SettingsManager.ts]: /gemini wdyt now?
